src/pages/workspaces/workspace/analysis/Analyses.ts

```diff
@@ -405,44 +406,61 @@ const Analyses = _.flow(
         div({
           style: { display: 'flex', alignItems: 'center' }
         }, [
-          label({ htmlFor: id, style: { fontWeight: 'bold', margin: '0 0.5rem', whiteSpace: 'nowrap' } }, 'Enable JupyterLab'),
-          h(Switch, {
-            id,
-            checked: enableJupyterLabGCP,
+          label({ htmlFor: id, style: { fontWeight: 'bold', margin: '0 0.5rem', whiteSpace: 'nowrap' } }, ['Enable JupyterLab']),
+          h(Switch as React.FC<any>, {
             onLabel: '', offLabel: '',
-            width: 40, height: 20,
             onChange: value => {
               setEnableJupyterLabGCP(value)
-              Ajax().Metrics.captureEvent(Events.analysisToggleJupyterLabGCP, { ...extractWorkspaceDetails(workspace.workspace), enabled: value })
-            }
-          })
+              setLocalPref(enableJupyterLabPersistenceId, value)
+              Ajax().Metrics.captureEvent(Events.analysisToggleJupyterLabGCP, { ...extractWorkspaceDetails(workspaceInfo), enabled: value })
+            },
+            id,
+            checked: enableJupyterLabGCP,
+            width: 40, height: 20,
+          }, [])
         ])
       ])])
     ])
   ])
 
+  const noAnalysisBanner: React.ReactElement<'div'> = div([
```

Comment:
I'll restate some previous feedback about not needing type declarations... The inferred type is adequate here and we don't really care that this is a div. It could easily be a different type of element.